### PR TITLE
Auto-refresh UserBar balance value

### LINKED ISSUE #41: [BUG] Real-time user's balance
**Describe the bug**

User's balance is not updating (`UserBar` component).

**To Reproduce**

1. Select an user.
2. Select a product.
3. Verify user's balance.
4. Buy!
5. Select same user again.

**Expected behavior**

Update user's balance after confirmation alert, maybe?

### PR DESCRIPTION
**Pull request by izn**
Hey ya.

This PR implements an auto-refresh in UserBar component.

Fixes #41.

# :camera: 

https://transfer.sh/kRmHF/RPReplay_Final1573582214.MP4

### CODE REVIEW

File: src/components/UserBar/UserBar.service.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+import { config } from '../../../config'
+
+export const getBalance = async (userName: string) => {
+  try {
+    const { data } = await axios.post(
+      `${config.GOOGLE_SCRIPT_URL}?action=balance`,
+      { user_name: userName },
+    )
+
+    return data.balance
```
Comment (rhogeranacleto) [inline src/components/UserBar/UserBar.service.ts]:
```suggestion
    return data.balance as number
```

File: src/components/UserBar/UserBar.component.tsx [hunk 2]
```diff
@@ -13,16 +35,18 @@ export class UserBar extends PureComponent<IUserBarProps> {
...
 
-  private getBalanceBadge(balance: number) {
-    return (
-      <Badge value={CurrencyHelper.parsePrice(balance)} status="primary" />
-    )
+  private getBalanceBadge() {
+    const { isLoading, balance } = this.state
+
+    return isLoading ?
+      <ActivityIndicator size="small" /> :
```
Comment (castilh0s) [inline src/components/UserBar/UserBar.component.tsx]: The ActivityIndicator component allow us to [change his color](https://facebook.github.io/react-native/docs/activityindicator.html), what do you think about set the same color as logo?

And we can just hide him changing the `animating` prop

Comment (castilh0s) [reply src/components/UserBar/UserBar.component.tsx]: Suggested change:

```
<ActivityIndicator size="small" color="pink" animating={isLoading}
```

Comment (castilh0s) [reply src/components/UserBar/UserBar.component.tsx]: With that we can remove the ternary expression

Comment (izn) [reply src/components/UserBar/UserBar.component.tsx]: Makes sense. You're right. But we still need a conditional to render the badge indicator, right? Still not sure if it's really worth rendering this component with implicit props. I think the ternary is explicit and cleaner. WDYT?

Anyway, I agree to change indicator color. :pineapple:

Comment (castilh0s) [reply src/components/UserBar/UserBar.component.tsx]: To badge we just can do `{!isLoading && <Badge />}`, WDYT?

Comment (izn) [reply src/components/UserBar/UserBar.component.tsx]: Nice one. I will wrap these components with a `<View>`.

File: src/components/UserBar/UserBar.component.tsx [hunk 3]
```diff
@@ -13,16 +36,35 @@ export class UserBar extends PureComponent<IUserBarProps> {
...
     )
   }
 
-  private getBalanceBadge(balance: number) {
+  private getBalanceBadge() {
+    const { isLoading, balance } = this.state
+
     return (
-      <Badge value={CurrencyHelper.parsePrice(balance)} status="primary" />
+      <ThemeConsumer>
```
Comment (izn) [inline src/components/UserBar/UserBar.component.tsx]: I think `ThemeConsumer` is good for now, but in future we could refactor to use React contexts/hooks.

Reference: https://react-native-elements.github.io/react-native-elements/docs/customization.html#using-the-theme-in-your-own-components

WDYT @castilh0s?

**Pull request merged**